inc/bluetooth/MicroBitUtilityService.h

```diff
@@ -0,0 +1,207 @@
+/*
+The MIT License (MIT)
+
+This class has been written for the Microbit Educational Foundation.
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+*/
+
+#ifndef MICROBIT_UTILITY_SERVICE_H
+#define MICROBIT_UTILITY_SERVICE_H
+
+#include "MicroBitConfig.h"
+
+#if CONFIG_ENABLED(DEVICE_BLE)
+
+#include "MicroBitBLEManager.h"
+#include "MicroBitBLEService.h"
+#include "MicroBitMemoryMap.h"
+
+#include "MicroBitFlash.h"
+#include "MicroBitStorage.h"
+
+#include "MicroBitComponent.h"
+#include "MicroBitEvent.h"
+#include "EventModel.h"
+#include "MicroBitLog.h"
+
+/* Set this to enable an alternative implementation, which creates a fiber to process requests.
+ * The default simply uses an event to trigger processing, and if processRequest() blocks,
+ * a temporary fiber is created as necessary.
+ */
+#ifndef MicroBitUtilityService_FIBER
+#define MicroBitUtilityService_FIBER 0
```

Comment:
Is there a reason to hold both these implementations in the final revision @martinwork? or shall we just pick one? (e.g. the event based approach?)

Reply:
Yes please pick one - whichever you think is the best idea.